### ISSUE TITLE
interceptor on service method

### ISSUE THREAD

**Issue by nguyeda** [nestjs/nest#1923]
## I'm submitting a...

<pre><code>
[ ] Regression 
[ ] Bug report
[ ] Feature request
[x] Documentation issue or request
[ ] Support request => Please do not submit support request here, instead post your question on Stack Overflow.
</code></pre>

## Current behavior
Hello, I'm discovering Nestjs and got a question about interceptors: is it possible to implement an interceptor on something that is not a controller ? I want to implement a custom cache using a decorator. I have a FooController that calls a FooService that calls a FooRepository. FooRepository calls a REST endpoint that give me limited rate and latency so I want to be able to cache that result. I can't do the caching a FooController level because FooService is creating a JWT token.

```
@Controller('foo')
export class FooController {
  constructor(private readonly fooService: FooService) {}
  @Get(':whatever')
  public async getFoo(@Param('whatever') whatever: string) {
    return this.fooService.getFoo(whatever);
  }
}

@Injectable()
export class FooService {
  constructor(private readonly fooRepository: FooRepository) {}
  public async getFoo(whatever: string) {
    const res = this.fooRepository.getFoo(whatever);
    // do some other things
    return res;
  }
}

@Injectable()
export class FooRepository {
  @Cached()
  public async getFoo(whatever: string) {
    return http.get(...)
  }
}
```

I currently use my own @Cached typescript decorator. 

```
export function Cached() {
  return (target: any, propertyKey: string, descriptor: PropertyDescriptor) => {
    const origin = descriptor.value;

    descriptor.value = async function(...args: any[]) {
      // check cache or call origin
      return origin.apply(this, args);
    }
  }
}
```

But what if I want to use nestjs resources there? (for instance an injectable redis client). I'm looking for something similar to the NestInterceptor implementation (https://docs.nestjs.com/interceptors).

**Comment by kamilmysliwiec** [comment]
> Is it possible to implement an interceptor on something that is not a controller

I believe that it was discussed a couple of times already. I'd suggest looking through existing, closed issues, you will surely find more detailed answer.

TL;DR - it's impossible
